### ISSUE TITLE
Support POST

### ISSUE THREAD

**Issue by kyleboyle** [evandroforks/PlantUmlConnection#1]
As implemented at:

https://github.com/plantuml/plantuml-server/pull/74

**Comment by evandrocoan** [comment]
The implementation here will not be straightforward because there, the code is implemented in the Java language, while here, it is in Python. I would not be implementing this soon because there is some time I do not use this. But contributions are welcome.

Right now, I am focused on finishing my bachelor thesis and next, on this other list of issues: https://github.com/evandrocoan/ITE/issues

**Comment by kyleboyle** [comment]
There is nothing to really implement except to issue a post instead of a get to the server. plantuml server has been updated to support POSTs so that there isn't a document length limit anymore.

eg here:
openurl = urlopen(request)

from urlopen docs:

If the url uses the http: scheme identifier, the optional data argument may be given to specify a POST request (normally the request type is GET).

I just wanted to raise the issue so that you / others are aware of the enhancement opportunity.

**Comment by kyleboyle** [comment]
Also I'm confused by your statement because you requested the POST feature here: https://github.com/plantuml/plantuml-server/issues/69

**Comment by evandrocoan** [comment]
This repository is only a wrapper which calls https://github.com/plantuml/plantuml-server/issues/69 to process the requests. In order for this to support POST requests, they should also be supported by them.

**Comment by kyleboyle** [comment]
Yes indeed. Merge request 74 adds that feature: plantuml/plantuml-server#74

**Comment by violetgo** [comment]
> There is nothing to really implement except to issue a post instead of a get to the server. plantuml server has been updated to support POSTs so that there isn't a document length limit anymore.
> 
> eg here:
> openurl = urlopen(request)
> 
> from urlopen docs:
> 
> If the url uses the http: scheme identifier, the optional data argument may be given to specify a POST request (normally the request type is GET).
> 
> I just wanted to raise the issue so that you / others are aware of the enhancement opportunity.

yes,I found the uml-file can't render if the file is large and use GET request.
I think maybe it should use POST request like this:

```
//plantumlconnection/all/plantuml_connection/__init__.py
class PlantUML(object):
    ...
    def processes(self, plantuml_text):
            ...
            request = Request(self.url)
            openurl = urlopen(request,plantuml_text)
```
